src/internal/stream/iterator.go

```diff
@@ -6,11 +6,20 @@ import (
 	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
 )
 
-// EOS signals the end of the stream
-var EOS = errors.New("end of stream")
+type ErrEndOfStream struct{}
```

Comment:
This probably doesn't need to be public, since we want people to use IsEOS and not a manual errors.Is.